test/tests.js

```diff
@@ -509,3 +585,128 @@ test("Tagging", function() {
   ok(decoded[2] instanceof SimpleValue, "third item is a SimpleValue");
   equal(decoded[2].value, 0xf0, "third item tag");
 });
+
+test("Encode Integer Keys as Integers", function() {
+  var good = 'a10a00';
+  var bad = 'a162313000';
+  var encoded = CBOR.encode({ 10: 0 });
+  var hex = arrayBufferToHex(encoded);
+  notEqual(hex, bad, "Encode integer string key not as string");
+  equal(hex, good, "Encode integer string key as int");
+});
+
+test("Throw when decoding nothing", function() {
+  var exc = false;
+  try {
+    CBOR.decode();
+  } catch ( err ) {
+    exc = true;
+  }
+  ok(exc, "Should throw when decoding nothing");
+});
+test("Should return undefined when passed empty buffer", function() {
+  var decoded = CBOR.decode(new ArrayBuffer(0));
+  equal(decoded, undefined, "decode empty as undefined");
+});
+
+test("Should throw when decoding nothing", function() {
+  var exc = false;
+  try {
+    CBOR.decode();
+  } catch ( err ) {
+    exc = true;
+  }
+  ok(exc, "Should throw when decoding nothing");
+});
+test("Should return undefined when passed empty buffer", function() {
+  var decoded = CBOR.decode(new ArrayBuffer(0));
+  equal(decoded, undefined, "decode empty as undefined");
+});
+
+test("Should encode would-be denormals some form correctly, not as denormals", function() {
+  var encoded, decoded;
+  encoded = CBOR.encode(1e-309);
+  ok(encoded, "encoding must produce result");
+  decoded = CBOR.decode(encoded);
+  equal(decoded, 1e-309, "decoded as expected");
+
+  encoded = CBOR.encode(-1e-309);
+  ok(encoded, "encoding must produce result");
+  decoded = CBOR.decode(encoded);
+  equal(decoded, -1e-309, "decoded as expected");
+
+  encoded = CBOR.encode(1e+309);
+  ok(encoded, "encoding must produce result");
+  decoded = CBOR.decode(encoded);
+  equal(decoded, 1e+309, "decoded as expected");
+
+  encoded = CBOR.encode(65504.00000000001);
+  ok(encoded, "encoding must produce result");
+  decoded = CBOR.decode(encoded);
+  equal(decoded, 65504.00000000001, "decoded as expected");
+
+  encoded = CBOR.encode(-65504.00000000001);
+  ok(encoded, "encoding must produce result");
+  decoded = CBOR.decode(encoded);
+  equal(decoded, -65504.00000000001, "decoded as expected");
+
+});
+
+test("Should throw when decoding invalid codes", function() {
+  var exc = false;
+  try {
+    CBOR.decode(hexToArrayBuffer('fc'));
+  } catch ( err ) {
+    exc = true;
+  }
+  ok(exc, "Should throw when decoding invalid codes");
+});
+/* no facility for testing; phantomjs does not have TextEncoder/TextDecoder/StringEncoder/StringDecoder
```

Comment:
need a facility that can cover performance enhancements involving node and browser specific features